back/src/Guide/Markdown.hs

```diff
@@ -318,7 +330,22 @@ instance Aeson.ToJSON MarkdownBlock where
     "html" Aeson..= utf8ToText (markdownBlockHtml md) ]
 instance Aeson.ToJSON MarkdownTree where
   toJSON md = Aeson.object [
-    "text" Aeson..= markdownTreeSource md ]
+    "text" Aeson..= markdownTreeSource md,
+    "prefix" Aeson..= markdownTreeIdPrefix md ]
+
+instance Aeson.FromJSON MarkdownInline where
+  parseJSON = Aeson.withObject "MarkdownInline" $ \o -> do
+    txt <- o Aeson..: "text"
+    pure $ toMarkdownInline txt
+instance Aeson.FromJSON MarkdownBlock where
+  parseJSON = Aeson.withObject "MarkdownBlock" $ \o -> do
+    txt <- o Aeson..: "text"
+    pure $ toMarkdownBlock txt
+instance Aeson.FromJSON MarkdownTree where
+  parseJSON = Aeson.withObject "MarkdownTree" $ \o -> do
+    txt <- o Aeson..: "text"
+    prefix <- o Aeson..:? "prefix" Aeson..!= T.empty
```

Comment:
You can use `""` instead of `T.empty`, I don't think there is a reason to not use overloaded strings here.